### DISCUSSION TITLE
@sentry/nextjs withSentryConfig enables productionBrowserSourceMaps, in the end, assets sourcemap is open to the public.

### DISCUSSION THREAD

**Discussion by seonghyeonkimm** [vercel/next.js#32920] [Help] [answered]
### Descriptions
- I am using `@sentry/nextjs` withSentryConfig to track errors in my production nextjs app.
- Recently, I found out that sentryConfig enables `productionBrowserSourceMaps` even if I did not set it to true in my next.config.js
- I understood that sentry needs assets' sourcemaps. However, I thought I need a way to distinguish build dir to upload to sentry release and build dir being served on my nextjs server.
- I don't think there is a way to do this right now in my knowledge. Anyone to have idea to handle this kind of problems? Please share with me.

**Comment by seonghyeonkimm** [comment]
For now, after build, I deleted `*.map` files manually. However, sourceMappingURL is already included in the assets files. I should see chrome devtool warning message saying **"it can't find sourcempas related to the asset".** I think it is quite annoying for me.

**Comment by andrewli-ca** [comment]
I'm facing the same issue where my code is being exposed in a `_N_E` folder under the source tab in Chrome dev tools.

I couldn't figure out a solution so what I end up doing is disabling the Sentry Webpack plugin for the client build. This completely disables source mapping on the client side and my source code is now hidden.

```js
// next.config.js
const moduleExports = {
  sentry: {
    disableServerWebpackPlugin: false,
    disableClientWebpackPlugin: true,
  },
};
```

This workaround works for me because my Next.js app uses SSR for all pages. I'm ok without having a stack trace If an unexpected error occurs on the client. This is not ideal but the tradeoff of hiding my source code exceeds not having a stack trace.

One thing to note is that I am using Next.js 11 and not hosting on Vercel. Maybe bumping up to Next.js 12 deploying to Vercel and using the [Vercel/Sentry Integration](https://docs.sentry.io/product/integrations/deployment/vercel/) might work. 🤷‍♂️

Check out the [docs](https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/#configure-source-maps) for more details on configuring the source map.

I also came across this related [GitHub issue](https://github.com/getsentry/sentry-webpack-plugin/issues/174). I have not tried the proposed solutions.

If you or anyone finds a solution, please share. 🙏

**Comment by seonghyeonkimm** [comment, marked as answer]
@andrewli-ca I just found that `@sentry/nextjs` is updated and started to support options `hideSourceMaps`. I think you can check [this pr](https://github.com/getsentry/sentry-javascript/pull/4436). Even if I didn't try to use it yet, I will do it soon.

**Reply by Yihao-G** [reply]
Just a heads up that this option only removes the `//# sourceMappingURL=` comment from the JS files. The `.js.map` files are still there and are accessible to the public if you visit the URL directly. You still need to manually remove the `.js.map` files from the final build.
